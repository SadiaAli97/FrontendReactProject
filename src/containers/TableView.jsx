import * as React from 'react';
import { Header } from '../components/Header';
import { SDKListItem } from '../components/SDKListItem';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import { ISDKS } from '../constants/app-constants';
import moment from 'moment';
import '../css/tableView.css';


export function TableView({data, selectedToggleOption}) {
  const headings = {
    heading: selectedToggleOption === ISDKS ? 'Installed SDK\'s' : 'Uninstalled SDK\'s',
    subheading: 'Latest Update: ' + moment().format('MMM Do YYYY'),
    count: data?.length
  }
  let categories = []
   data.map(item => {
    const allItems = [...item.categories];
    allItems.map(i => {
      if (!categories.includes(i)) {
        categories.push(i);
      }
    })
  });
  return (
    <div className='table-outline'>
      <Header
        {...headings}
      />
      <div className='divider' />
      <Box sx={{ flexGrow: 1 }}>
      <Grid container spacing={2}>
      {categories?.map(category => {
        return <Grid item xs={6}>
          <SDKListItem
            sdkCategory={category}
            categoryItems={data.filter(row => row.categories.includes(category))}
          />
        </Grid>
      })}
      </Grid>
    </Box>
    </div>
  );
}