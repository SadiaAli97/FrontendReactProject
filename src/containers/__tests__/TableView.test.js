import TestRenderer from 'react-test-renderer';
import { TableView } from '../TableView';

it('renders TableView ', () => {
	const testRenderer = TestRenderer.create(
		<TableView
		  data={[{
        "id": "1",
        "name": "SDK 1",
        "categories": ["Backend"],
        "firstSeenDate": "2017-11-16T10:48:53.000-08:00",
        "lastSeenDate": "2019-07-16T13:24:39.000-07:00",
        "__typename": "Sdk"
      },]}
		  selectedToggleOption='isdks'
		/>
	);
  expect(testRenderer.toJSON()).toMatchSnapshot()
});

