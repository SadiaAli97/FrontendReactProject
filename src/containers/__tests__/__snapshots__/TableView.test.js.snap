// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TableView  1`] = `
<div
  className="table-outline"
>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
      >
        <div
          className="heading"
        >
          Installed SDK's
        </div>
        <div
          className="subheading"
        >
          Latest Update:
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
      >
        <div
          className="listcount"
        >
          1
        </div>
      </div>
    </div>
  </div>
  <div
    className="divider"
  />
  <div
    className="MuiBox-root css-i9gxme"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
      >
        <div
          className="list-item"
        >
          <div
            className="category"
          >
            Backend
          </div>
          <div
            className="items"
          >
            <div
              className="item"
            >
              SDK 1 2019-07-16T13:24:39.000-07:00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
